display/d.legend/draw.c

```diff
@@ -323,13 +323,20 @@ void draw(const char *map_name, int maptype, int color, int thin, int lines,
             }
         }
 
-        /* In case of log. scale raster doesn't contain negative or zero values
-         */
-        if (log_sc)
-            if ((dmin <= 0) || (dmax <= 0))
-                G_fatal_error(
-                    _("Range [%.3f, %.3f] out of the logarithm domain."), dmin,
-                    dmax);
+        if (log_sc) {
+            if (dmax <= 0)
+                G_fatal_error(_("No positive cell values found; "
+                                "logarithmic legend cannot be displayed"));
+            if (dmin > 0)
+                eps_min = dmin;
+            else {
+                G_warning(_("Non-positive cell values found; "
+                            "discarding colors for cells <= %g and "
+                            "assuming a positive min cell value of %g"),
+                          eps, eps);
+                eps_min = eps;
```

Comment:
I wonder if you couldn't just use `dmin = eps_min` here, which would avoid all the changes below.